src/algorithms/SortingAlgorithms.java

```diff
@@ -1,47 +1,189 @@
+                }
+            }
+        }
+        System.out.print("Array was sorted by using \"Insertion Sorting\" method --> ");
+        print(array);
     }
 
-    public static void sortByMergeSorting(int[] array){
+    public static void sortByMergeSorting(int[] array) {
+        if (array.length < 2)
+            return;
+        int halfLength = array.length / 2;
+        int[] leftHalfOfArray = new int[halfLength];
+        int[] rightHalfOfArray = new int[array.length - halfLength];
+        for (int i = 0; i < leftHalfOfArray.length; i++) {
+            leftHalfOfArray[i] = array[i];
+        }
+        for (int i = 0; i < rightHalfOfArray.length; i++) {
+            rightHalfOfArray[i] = array[leftHalfOfArray.length + i];
+        }
+        sortByMergeSorting(rightHalfOfArray);
+        sortByMergeSorting(leftHalfOfArray);
+        merge(array, leftHalfOfArray, rightHalfOfArray);
+    }
 
+    public static void sortByQuickSorting(int[] array) {
+        if (array.length <= 1)
+            return;
+        int index = 0;
+        int pivot = array[index];
+        for (int i = 0; i < array.length; ) {
+            for (int j = array.length - 1; j > 0; j--) {
+                if (pivot > array[j]) {
+                    swap(array, index, j);
+                    index = j;
+                    break;
+                }
+            }
+            int pivotIndex = index;
+            while (i < pivotIndex) {
+                if (pivot > array[i]) {
+                    i++;
+                } else {
+                    swap(array, index, i);
+                    index = i;
+                }
+                while (index == i && i < pivotIndex) {
+                    if (pivot < array[pivotIndex]) {
+                        pivotIndex--;
+                    } else if (pivot > array[pivotIndex]) {
+                        swap(array, index, pivotIndex);
+                        index = pivotIndex;
+                        i++;
+                    }
+                }
+            }
+            pivot = array[index];
+            break;
+        }
+        int[] leftHalfOfArray = new int[array.length - index - 1];
+        int[] rightHalfOfArray = new int[array.length - leftHalfOfArray.length];
+        for (int i = 0; i < leftHalfOfArray.length; i++) {
+            leftHalfOfArray[i] = array[i];
+        }
+        for (int i = 0; i < rightHalfOfArray.length; i++) {
+            rightHalfOfArray[i] = array[leftHalfOfArray.length + i];
+        }
+        sortByQuickSorting(leftHalfOfArray);
+        sortByQuickSorting(rightHalfOfArray);
     }
 
-    public static void sortByQuickSorting(int[] array){
+    private static void merge(int[] destArray, int[] arrayOne, int[] arrayTwo) {
+        int i = 0;
+        int j = 0;
+        int k = 0;
+        while (i < arrayOne.length && j < arrayTwo.length) {
+            if (arrayOne[i] <= arrayTwo[j]) {
+                destArray[k] = arrayOne[i];
+                i++;
```

Comment:
In this method you can easily use POST-INCREMENT inside the brackets.

For example: **destArray[k++] = arrayOne[i++];**